src/daft-dsl/src/functions/python/udf_runtime_binding.rs

```diff
@@ -0,0 +1,72 @@
+use std::hash::{Hash, Hasher};
+
+use serde::{de::Visitor, Deserialize, Serialize};
+
+use super::RuntimePyObject;
+
+/// A binding between the StatefulPythonUDF and an initialized Python callable
+///
+/// This is `Unbound` during planning, and bound to an initialized Python callable
+/// by an Actor right before execution.
+///
+/// Note that attempting to Hash, Eq, Serde this when it is bound will panic!
+#[derive(Debug, Clone)]
+pub enum UDFRuntimeBinding {
```

Comment:
Instead of binding UDFs and then having to implement these various structs in order to be inside of an expression, we could separate out the stateful and stateless parts of the projection to run separately.

Reply:
I think we can try this approach as a follow-on, and it shouldn't block merging of this PR for now